src/snmalloc/pal/pal_linux.h

```diff
@@ -232,6 +240,50 @@ namespace snmalloc
       // its APIs are not exception-free.
       return dev_urandom();
     }
+
+#  ifdef SNMALLOC_HAS_LINUX_FUTEX_H
+    using WaitingWord = int;
+
+    template<class T>
+    static void wait_on_address(std::atomic<T>& addr, T expected)
+    {
+      static_assert(
+        sizeof(T) == sizeof(WaitingWord) && alignof(T) == alignof(WaitingWord),
+        "T must be the same size and alignment as WaitingWord");
+      for (;;)
+      {
+        if (addr.load(std::memory_order_relaxed) != expected)
+          break;
+
+        long ret = syscall(
+          SYS_futex, &addr, FUTEX_WAIT_PRIVATE, expected, nullptr, nullptr, 0);
+
+        if (ret == -EINTR)
```

Comment:
Why the minus EINTR?
```suggestion
        if (ret == -1 && errno == EINTR)
```